lib/internal/modules/esm/hooks.js

```diff
@@ -154,7 +155,26 @@ class Hooks {
       const next = this.#chains.load[this.#chains.load.length - 1];
       ArrayPrototypePush(this.#chains.load, { __proto__: null, fn: load, url, next });
     }
-    return initialize?.(data);
+
+    const hooks = await initialize?.(data);
+ 
+    if (hooks?.resolve) {
+      if (resolve) {
+        throw new Error('Invariant violation: both a module-level and returned resolve hook found');
+      }
```

Comment:
I threw this in as a placeholder to guard against the element of surprise. Not attached to this thought it would change the paradigm such that one `register()` could produce multiple hooks in the chain.

Reply:
Swapped to `ERR_INTERNAL_ASSERTION`.